### PR TITLE
fix: use web standard crypto api

### PR DESCRIPTION
**Pull request by himself65**

### CODE REVIEW

File: packages/waku/src/lib/builder/build.ts [hunk 1]
```diff
@@ -1,46 +1,44 @@
-import { createHash } from 'node:crypto';
 import { Readable } from 'node:stream';
```
Comment (himself65) [inline packages/waku/src/lib/builder/build.ts:1]: IDK why I ran `prettier --write .` and order totally changed

File: packages/waku/src/lib/builder/build.ts [hunk 2]
```diff
@@ -76,17 +74,16 @@ const onwarn = (warning: RollupLog, defaultHandler: LoggingFunction) => {
...
-    const sha256 = createHash('sha256');
-    sha256.on('readable', () => {
-      const data = sha256.read();
-      if (data) {
-        resolve(data.toString('hex').slice(0, 9));
-      }
-    });
-    createReadStream(fname).pipe(sha256);
-  });
+const hash = async (fname: string): Promise<string> => {
```
Comment (himself65) [inline packages/waku/src/lib/builder/build.ts:77]: https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest#converting_a_digest_to_a_hex_string

File: packages/waku/src/lib/builder/build.ts (hunk 1 above)
Comment (dai-shi) [reply packages/waku/src/lib/builder/build.ts:1]: You are probably using a different version or a different config.

File: packages/waku/src/lib/builder/build.ts [hunk 3]
```diff
@@ -76,17 +74,16 @@ const onwarn = (warning: RollupLog, defaultHandler: LoggingFunction) => {
...
+  const encoder = new TextEncoder();
+  const data = encoder.encode(fname);
```
Comment (dai-shi) [inline packages/waku/src/lib/builder/build.ts:79]: This seems wrong. We need to create a hash from the file content, not the file name.

File: packages/waku/src/lib/builder/build.ts (hunk 1 above)
Comment (dai-shi) [reply packages/waku/src/lib/builder/build.ts:1]: Hope you figure it out...

**Review by dai-shi — CHANGES_REQUESTED** [review]

**Review by dai-shi — APPROVED** [review]
LGTM
Ready to merge?

**Comment by himself65** [comment]
Yes

File: packages/waku/src/lib/builder/build.ts [hunk 4]
```diff
@@ -98,6 +84,7 @@ const analyzeEntries = async (
   );
   const clientFileSet = new Set<string>([wakuClientDist]);
   const serverFileSet = new Set<string>();
+  const fileHashSet = new Map<string, string>();
```
Comment (dai-shi) [inline packages/waku/src/lib/builder/build.ts:87]:
```suggestion
  const fileHashMap = new Map<string, string>();
```

**Review by dai-shi — CHANGES_REQUESTED** [review]
No clue for prettier config?

**Review by dai-shi — APPROVED** [review]

**Pull request merged**